app/services/data_stage/update_staged_schools.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+module DataStage
+  class UpdateStagedSchools < ::BaseService
+    include GiasTypes
+
+    def initialize(school_data_file:, school_links_file:)
+      @school_data_file = school_data_file
+      @school_links_file = school_links_file
+    end
+
+    def call
+      update_schools
+      update_school_links
+    end
+
+  private
+
+    def update_schools
+      if @school_data_file
+        CSV.foreach(@school_data_file, headers: true, encoding: "ISO-8859-1:UTF-8") do |row|
+          if eligible_row? row
+            school_attrs = filtered_attributes(row)
+            urn = school_attrs.fetch(:urn)
+
+            school = DataStage::School.find_by(urn: urn)
+
+            if school.present?
+              school.update!(school_attrs)
+            else
+              DataStage::School.create!(school_attrs)
+            end
+          end
+        end
+      end
+    end
+
+    def update_school_links
+      if @school_links_file
+        CSV.foreach(@school_links_file, headers: true, encoding: "ISO-8859-1:UTF-8") do |row|
+          link_attrs = link_attributes(row)
+
+          school = DataStage::School.find_by(urn: link_attrs[:urn])
+
+          if school.present?
+            link = school.school_links.find_by(link_urn: link_attrs[:link_urn])
+
+            if link.present?
+              if link.link_type != link_attrs[:link_type]
+                link.update!(link_type: link_attrs[:link_type])
+              end
+            else
+              school.school_links.create!(link_attrs.except(:urn))
+            end
+          end
+        end
+      end
+    end
+
+    def eligible_row?(row)
+      english_district_code?(row.fetch("DistrictAdministrative (code)")) &&
+        eligible_establishment_code?(row.fetch("TypeOfEstablishment (code)").to_i)
+    end
+
+    def filtered_attributes(data)
```

Comment:
Feels like this looks very similar to the school importer, and for good reason too, they do the same job. Is this going to essentially replace it? If not, should we commonise some things between them?

Reply:
So, currently this is a refactored version of the school importer. I split fetching the files out from the actual updating to simplify things, plus I don't want to change the existing one currently so that we can still update schools directly, until this makes it out properly.  Eventually this should replace the importer as it will no longer be needed.